### PR TITLE
 #667 Error setting resourceFile: "Maximum call stack size exceeded"

### PR DESCRIPTION
**Pull request by cima**
+ Preventing the infinite recursion by not translating warning about missing translation.

Please see and reopen issue #667 . What I provide here is fix of catastrphic failure in infinite recursion. It doesn't fix translation in compilation systems like webpack.

### CODE REVIEW

File: node/internal.ts [hunk 1]
```diff
@@ -200,10 +200,10 @@ export function _loc(key: string, ...param: any[]): string {
     }
     else {
         if (Object.keys(_resourceFiles).length <= 0) {
-            _warning(_loc('LIB_ResourceFileNotSet', key));
+            _warning(`Resource file haven\\'t been set, can\\'t find loc string for key: ${path}`);
```
Comment (anatolybolshakov) [inline node/internal.ts:203]:
```suggestion
            _warning(`Resource file haven\'t been set, can\'t find loc string for key: ${path}`);
```

File: node/internal.ts [hunk 2]
```diff
@@ -200,10 +200,10 @@ export function _loc(key: string, ...param: any[]): string {
...
         }
         else {
-            _warning(_loc('LIB_LocStringNotFound', key));
+            _warning(`Can\\'t find loc string for key: ${path}`);
```
Comment (anatolybolshakov) [inline node/internal.ts:206]:
```suggestion
            _warning(`Can\'t find loc string for key: ${path}`);
```

**Comment by anatolybolshakov** [comment]
@cima thanks for the fix here!
Could you please take a look at suggestions above and bump patch number of package version also?

**Comment by cima** [comment]
@anatolybolshakov I have taken a look at it and I wasn't happy with my own code. I've just add a brief unit test to be able to put breakpoint and debug and discovered two things:

1. Backslash is not necessary at all as it is in backtics that translates to double quotes so the direct apostrophe is not interfering with compiler when parsing a string quotation.
2. The parameter in formated string was totally wrong. It supposed to be `key` not `path`.

I have compile it, run tests and the fix is pushed in the branch.

File: node/test/internalhelpertests.ts [hunk 3]
```diff
@@ -401,4 +401,10 @@ describe('Internal Path Helper Tests', function () {
 
         done();
     });
+    
+    it('ReportMissingLocalization', (done: MochaDone) => {
```
Comment (anatolybolshakov) [inline node/test/internalhelpertests.ts:405]: @cima this test does not really new changes - it passes successfully with and without new changes.

Comment (cima) [reply node/test/internalhelpertests.ts:405]: You are right that the test finishes OK regardles off my change. The propper testing would require some serious mocking using rewire mock, which would require new dependencies. Maybe damaging `__dirname` variable within the unittest might do the trick. It is bit strettching beyond replacing two strings. Shall I fully simulate the situation when no file is loaded?

Comment (anatolybolshakov) [reply node/test/internalhelpertests.ts:405]: @cima yes, I guess that would be more helpful for coverage purposes.

Comment (cima) [reply node/test/internalhelpertests.ts:428]: I have included a test that simulates the missing file. When such test (5da2a2a98428d6300bde4e529237ad181c1bd2a1) is applied to master (ba1f94be19f958f1f718ac946aef3123c8eee4a8) it triggers the infinite recusion.

Comment (anatolybolshakov) [reply node/test/internalhelpertests.ts:428]: @cima thanks a lot!

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks for contribution!

**Review by AnnaOpareva — APPROVED** [review]

**Pull request merged**